pkg/policy/api/rule_validation.go

```diff
@@ -359,17 +360,25 @@ func (pr *PortRule) sanitize(ingress bool) error {
 		if isZero {
 			haveZeroPort = true
 		}
-		hasDNSRules := pr.Rules != nil && len(pr.Rules.DNS) > 0
 		// DNS L7 rules can be TCP, UDP or ANY, all others are TCP only.
 		switch {
 		case pr.Rules.IsEmpty(), hasDNSRules:
 			// nothing to do if no rules OR they are DNS rules (note the comma above)
 		case pr.Ports[i].Protocol != ProtoTCP:
 			return fmt.Errorf("L7 rules can only apply to TCP (not %s) except for DNS rules", pr.Ports[i].Protocol)
 		}
+	}
+
+	if ingress && hasDNSRules {
+		return fmt.Errorf("DNS rules are not allowed on ingress")
+	}
 
-		if ingress && hasDNSRules {
-			return fmt.Errorf("DNS rules are not allowed on ingress")
+	if len(pr.ServerNames) > 0 && !pr.Rules.IsEmpty() && pr.TerminatingTLS == nil {
```

Comment:
Ditto.

Reply:
Moved, although it is a matter of taste as in the normal non-error case all these checks will be executed anyway, so there is really no point trying to optimize the order of these checks.